File: php/chainofresponsibility/Leader.php
<?php

abstract class Leader
{
    protected $name;
    // 责任链后继对象
    protected $nextLeader;

    public function __construct($name)
    {
        $this->name = $name;
    }

    public function getNextLeader()
    {
        return $this->nextLeader;
    }

    /**
     * 设置责任链后继对象
     * @param [type] $nextLeader [description]
     */
    public function setNextLeader($nextLeader)
    {
        $this->nextLeader = $nextLeader;
    }

    public abstract function handleRequest($request);
}

File: java/pattern/src/chainofresponsibility/Leader.java
package chainofresponsibility;

/**
 * 抽象类
 */
public abstract class Leader {
    protected String name;
    // 责任链后继对象
    protected Leader nextLeader;

    public Leader(String name) {
        this.name = name;
    }

    public Leader getNextLeader() {
        return nextLeader;
    }
    // 设置责任链后继对象
    public void setNextLeader(Leader nextLeader) {
        this.nextLeader = nextLeader;
    }

    // 处理请求的核心的业务方法
    public abstract void handleRequet(LeaveRequest request);
}
